homeassistant/components/binary_sensor/google_calendar.py

```diff
@@ -0,0 +1,224 @@
+"""
+Support for custom shell commands to to retrieve values.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/binary_sensor.command/
+"""
+# pylint: disable=import-error
+import logging
+from datetime import timedelta
+from homeassistant.components.binary_sensor import BinarySensorDevice
+from homeassistant.util import Throttle
+
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'google_calendar'
+ENTITY_ID_FORMAT = DOMAIN + '.{}'
+
+CONF_NAME = 'name'
+CONF_TRACK = 'track'
+CONF_SEARCH = 'search'
+CONF_OFFSET = 'offset'
+
+DEFAULT_CONF_TRACK_NEW = True
+DEFAULT_CONF_OFFSET = '#-'
+
+# Return cached results if last scan was less then this time ago
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=5)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Z-Wave platform for sensors."""
+    if discovery_info is None:
+        return
+
+    add_devices([GoogleCalendarSensor(hass, discovery_info['cal_id'], data)
+                 for data in discovery_info['entities'] if data['track']])
+
+
+# pylint: disable=too-many-instance-attributes
+class GoogleCalendarSensor(BinarySensorDevice):
+    """A calendar binary sensor.
+
+    It enables/disabled based on:
+     - Search criteria
+     - all events on a Google calendar.
+
+    """
+
+    # pylint: disable=too-many-arguments
+    def __init__(self, hass, calendar_id, data):
+        """Create the Google Calendar Sensor."""
+        from homeassistant.helpers.entity import generate_entity_id
+        self.hass = hass
+        self._calendar_id = calendar_id
+        self._track = data.get('track', DEFAULT_CONF_TRACK_NEW)
+        self._search = data.get('search', None)
+        self._name = data.get('name')
+        self._all_day = False
+        self._state = False
+        self._offset = data.get('offset', DEFAULT_CONF_OFFSET)
+        self._offset_time = 0
+        self.entity_id = generate_entity_id(ENTITY_ID_FORMAT,
```

Comment:
Actually it's not. It's defined in Entity and only used in the template sensor which does the same thing I'm doing, almost.

```
mnestor@ha:~/home-assistatant/homeassistant# grep --include \*.py -R generate_entity_id
helpers/entity.py:def generate_entity_id(entity_id_format, name, current_ids=None, hass=None):
components/binary_sensor/google_calendar.py:        from homeassistant.helpers.entity import generate_entity_id
components/binary_sensor/google_calendar.py:        self.entity_id = generate_entity_id(ENTITY_ID_FORMAT,
components/binary_sensor/template.py:from homeassistant.helpers.entity import generate_entity_id
components/binary_sensor/template.py:        self.entity_id = generate_entity_id(ENTITY_ID_FORMAT, device,
```